clients/common/utility.cpp

```diff
@@ -57,6 +57,16 @@ double get_time_us_sync(hipStream_t stream)
     return (tv.tv_sec * 1000 * 1000) + tv.tv_usec;
 };
 
+/* ============================================================================================ */
+bool is_limited_memory(size_t needed_mem)
+{
+    size_t     free_mem  = 0;
+    size_t     total_mem = 0;
+    hipError_t err       = hipMemGetInfo(&free_mem, &total_mem);
+
```

Comment:
What is the cost of this in terms of time?     Wondering if it is feasible to just move this into the device* allocators and return a failure that is handled by the CHECK_HIP_ERROR on device memory rather than all these special cases?

Reply:
You could also replace the CHECK_HIP_ERROR calls with a CHECK_HIP_DEV_ALLOC if hipMalloc failure is fine to trigger without checking memory free.